src/StatusStreamInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace CascadeEnergy\StatusStream;
+
+interface StatusStreamInterface
+{
+    /**
+     * Updates the current state to `active`, possibly with some context data
+     *
+     * @param mixed|null $context
+     */
+    public function active($context = null);
+
+    /**
+     * Updates the current state to `degraded`, possibly with some context data
```

Comment:
Operating in a way that is not ideal, but recoverable (e.g. if the digest system tried to process data for a measurable but the measurable was locked by another digest process that might be a temporary "degraded" state -- it's normal and recoverable, but means that the process is delayed).